File: easy/2710_remove_zeros_from_string.cpp
#include <string>
#include <stdio.h>

class Solution {
public:
    //std::string removeTrailingZeros(std::string num) {
    //    int len = num.size();
    //    int i = len - 1;
    //    while (num.at(i) == '0') {
    //        i--;
    //        num.pop_back();
    //    }
    //    return num;
    //}

    std::string removeTrailingZeros(std::string num) {
        while (num.back() == '0' && num.size()) {
            num.pop_back();
        }
        return num;
    }
};

File: easy/2710_remove_zeros_from_string.py
class Solution:
    def removeTrailingZeros(self, num: str) -> str:
        if num[-1] != "0":
            return num
        i = -2
        while num[i] == '0':
            i-=1
        return num[:i + 1]
    
s = Solution()
print(s.removeTrailingZeros("5123010000"))